Mathlib/Tactic/Linter/TerminalRefineLinter.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+#  The "terminal refine" linter
+
+The "terminal refine" linter flags terminal usages of `refine`.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter
+
+/-- The terminal refine linter emits a warning on terminal usages of `refine`. -/
+register_option linter.terminalRefine : Bool := {
+  defValue := true
```

Comment:
I light of #15616 this should be `false` and the linter added to the mathlib-only linters.